### PR TITLE
Parallel tests

### PR DESCRIPTION
**Pull request by johmathe**
## Checklist

- [ ] My pull request has a clear and explanatory title.
- [ ] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [ ] I added appropriate unit tests.
- [ ] I made sure the code passes all unit tests. (refer to comment below)
- [ ] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [ ] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [ ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [ ] I linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

**Comment by luisfpereira** [comment]
@johmathe, do you have any idea why this doesn't work out?

**Comment by johmathe** [comment]
Hard to parse from the logs here - it probably needs a few hours of investigation to see if it's possible to reproduce locally. I'm currently off for a week so hard to dig until July 6th :)
